src/pages/EventPage/EventPage.tsx

```diff
@@ -88,34 +91,48 @@ export default function EventPage({
         };
     };
 }): JSX.Element {
+    const userRecord = useUserRecordContext();
     const [modalShow, setModalShow] = React.useState<boolean>(false);
     const [heatMapData, setHeatMapData] = React.useState<HeatMapData>();
 
     React.useEffect(() => {
-        return getDocSnapshot$(`/events/${match.params.id}`, {
-            next: (snapshot) => {
-                const event = snapshot.data() as EventData;
-                const tempYTimes = getYTimesSorted(event.availability);
-                const tempXDays = getXDaysSorted(
-                    tempYTimes,
-                    event.availability
-                );
-                setHeatMapData({
-                    yData: tempYTimes,
-                    xData: formatXDays(tempXDays),
-                    mapData: createAvailabilityDataArray(
+        if (userRecord) {
+            getDocSnapshot$(`/events/${match.params.id}`, {
+                next: (eventSnapshot) => {
+                    const event = eventSnapshot.data() as EventData;
+                    const tempYTimes = getYTimesSorted(event.availability);
+                    const tempXDays = getXDaysSorted(
                         tempYTimes,
-                        tempXDays,
                         event.availability
-                    ),
-                    zeroState: createZeroStateArray(
-                        tempYTimes.length,
-                        tempXDays.length
-                    ),
-                });
-            },
-        });
-    }, []);
+                    );
+                    const formatedTempXDays = formatXDays(tempXDays);
+                    const zeroState = createZeroStateArray(
+                        LABELS.yLabels.length,
+                        LABELS.xLabels.length
+                    );
```

Comment:
Should the zeroState be sized to the user defaults here? Was there an issue with how they were written before? If possible I it should be consistent with how we're populating the preloadData from createPreloadArray.

Reply:
And you're right, the zero state size should reflect the group availability, I'll change that.